libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -976,18 +976,33 @@ private void updateDataFromBannerText(@NonNull BannerText primaryBannerText, Ban
   }
 
   private void loadPrimary(BannerText primaryBannerText) {
-    stepPrimaryText.setMaxLines(2);
+    if (!ViewUtils.isLandscape(getContext())) {
+      stepPrimaryText.setMaxLines(2);
+    }
     stepSecondaryText.setVisibility(GONE);
     adjustBannerTextVerticalBias(0.5f);
+
+    if (ViewUtils.isLandscape(getContext())) {
+      stepPrimaryText.setMaxWidth(instructionLayoutText.getWidth());
+    }
     loadTextWith(primaryBannerText, stepPrimaryText);
   }
 
   private void loadPrimaryAndSecondary(BannerText primaryBannerText, BannerText secondaryBannerText) {
     stepPrimaryText.setMaxLines(1);
     stepSecondaryText.setVisibility(VISIBLE);
     adjustBannerTextVerticalBias(0.65f);
+
+    if (ViewUtils.isLandscape(getContext())) {
+      int primaryTextMaxWidth = (int) (instructionLayoutText.getWidth() * 0.75);
```

Comment:
Magic Number 0.75. Can we extract it out in a separate constant, so that it is easy to understand it's need?

Reply:
yes, it's a magic number indeed. I can update it.